pkg/kubelet/kubelet_pods.go

```diff
@@ -1249,40 +1247,55 @@ func (pk *podKillerWithChannel) checkAndMarkPodPendingTerminationByPod(podPair *
 		} else {
 			klog.V(4).Infof("running pod %q is pending termination", podPair.RunningPod.ID)
 		}
-		return true
 	}
-	return false
 }
 
 // Close closes the channel through which requests are delivered
 func (pk *podKillerWithChannel) Close() {
 	close(pk.podKillingCh)
 }
 
-// KillPod sends pod killing request to the killer
+// KillPod sends pod killing request to the killer after marks the given pod
 func (pk *podKillerWithChannel) KillPod(pair *kubecontainer.PodPair) {
+	pk.markPodPendingTerminationByPod(pair)
 	pk.podKillingCh <- pair
 }
 
 // PerformPodKillingWork launches a goroutine to kill a pod received from the channel if
 // another goroutine isn't already in action.
 func (pk *podKillerWithChannel) PerformPodKillingWork() {
+	killing := sets.NewString()
+	var lock sync.Mutex
 	for podPair := range pk.podKillingCh {
-		if pk.checkAndMarkPodPendingTerminationByPod(podPair) {
```

Comment:
checkAndMarkPodPendingTerminationByPod adds the pod to the killing mutex... Adding the mutex and set back to this function is effectively the same functionality as checkAndMarkPodPendingTerminationByPod

Reply:
Ahh. This synchronizes the killing goroutine on 1289.

Reply:
We don't have to check it here anymore because I have limited to one request per pod in buffered channel.